logins-sql/src/schema.rs

```diff
@@ -0,0 +1,202 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use error::*;
+use db;
+
+// Note: care is taken to be compatible with the iOS database format.
+// (It's not clear if this is actually worthwhile though...)
+const VERSION: i64 = 4;
+
+pub const MIRROR_TABLE_NAME: &'static str = "loginsM";
+pub const LOCAL_TABLE_NAME: &'static str = "loginsL";
+pub const META_TABLE_NAME: &'static str = "loginsSyncMeta";
+
+const IDX_MIRROR_OVERRIDEN_HOSTNAME: &'static str = "idx_loginsM_is_overridden_hostname";
+const IDX_LOCAL_DELETED_HOSTNAME: &'static str = "idx_loginsL_is_deleted_hostname";
+
+// Every column shared by both tables except for `id`
+pub const COMMON_COLS: &'static str = "
+    guid,
+    username,
+    password,
+    hostname,
+    httpRealm,
+    formSubmitURL,
+    usernameField,
+    passwordField,
+    timeCreated,
+    timeLastUsed,
+    timePasswordChanged,
+    timesUsed
+";
+
+const COMMON_SQL: &'static str = "
+    id                  INTEGER PRIMARY KEY AUTOINCREMENT,
+    hostname            TEXT NOT NULL,
+    httpRealm           TEXT,
+    formSubmitURL       TEXT,
+    usernameField       TEXT,
+    passwordField       TEXT,
+    timesUsed           INTEGER NOT NULL DEFAULT 0,
+    -- Milliseconds
```

Comment:
A comment on why milliseconds (given how surprising it was to us both!) would be helpful here.